### PR TITLE
Make opened file visible in lookup

### PR DESCRIPTION
**Pull request by monthonk**
We have implemented this in readdir but missed it in lookup. We want the new files to be visible there as well. There might be two options here, 1/ We can lookup locally first and return immediately when the item we want is an uncommitted inode and we don't have to call any S3 APIs since it should not exist in S3 yet, or 2/ we can lookup remotely first then fall back to local. In this commit, we go with the second option because there is a possibility that someone could make a change remotely like creating a directory marker that shadow the uncommitted file.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I agree to the terms of the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).

### CODE REVIEW

File: mountpoint-s3/src/inode.rs [hunk 1]
```diff
@@ -238,7 +238,27 @@ impl Superblock {
...
+                InodeKindData::File { .. } => unreachable!("we know this is a directory"),
+                InodeKindData::Directory {
+                    children,
+                    writing_children,
+                } => {
+                    if let Some(inode) = children.get(name) {
+                        if writing_children.contains(&inode.ino()) {
+                            let inode = inode.clone();
+                            let stat = inode.inner.sync.read().unwrap().stat.clone();
+                            Ok(LookedUp { inode, stat })
```
Comment (passaro) [inline mountpoint-s3/src/inode.rs:253]: nit: maybe return here and just fallback to Err otherwise, avoiding the multiple else branches below.

**Review by passaro — APPROVED** [review]
With option 2, this is consistent with readdir, right? 
While writing to a new file:
* it will be visible in lookup and readdir,
* but if a file with the same key is added remotely, it will shadow the local one (at least `finish_writing`)
Do we have a way to test this behaviour?

File: mountpoint-s3/src/inode.rs [hunk 2]
```diff
@@ -238,7 +238,27 @@ impl Superblock {
                 .update_or_insert(parent_ino, name, stat.clone(), InodeKind::File)?;
             Ok(LookedUp { inode, stat })
         } else {
-            Err(InodeError::FileDoesNotExist)
```
Comment (sauraank) [inline mountpoint-s3/src/inode.rs:241]: Should we also add this case in the comments at line 233, otherwise comment can be misleading?

Comment (monthonk) [reply mountpoint-s3/src/inode.rs:241]: Updated, I think we can say on line 233 that the object doesn't exist remotely.

**Comment by monthonk** [comment]
@passaro yes, I think it's consistent with readdir and it's possible to write that kind of test maybe in `write_test.rs`. But I'm not sure if we want to, since we will just return remote result first, whatever it might be.

**Review by passaro — APPROVED** [review]
LGTM

**Pull request merged**